src/CtxHooks.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+import * as React from 'react';
+
+class CtxStore {
+  state = {}
+  subs = [];
+
+  subscribe(handler: Function) {
+    const wrapper = () => {
+      handler(this.state);
+    };
+
+    this.subs.push(wrapper);
+
+    const unsub = () => {
+      const index = this.subs.indexOf(handler);
+
+      if (index !== -1) {
+        this.subs.splice(index, 1);
+      }
+    };
+
+    return unsub;
+  }
+
+  replaceStateAndNotify(newState: any) {
+    this.state = newState;
+    this.subs.forEach(sub => sub());
+  }
+
+  getState() {
+    return this.state;
+  }
+}
+
+function createUseSelector(Context) {
+  return function useSelector(selector) {
+    const store = React.useContext(Context);
+    const [, forceUpdate] = React.useState(0);
+    let selectedState;
+
+    React.useEffect(() => {
+      const unsub = store.subscribe(selector);
+      forceUpdate(x => !x);
```

Comment:
this should be:

```js
const unsub = store.subscribe((state) => {
  if (selector produces a different value from last time) {
    forceUpdate(x => !x);
  }
});
```

(point being that the forceUpdate happens inside the subscribe callback)